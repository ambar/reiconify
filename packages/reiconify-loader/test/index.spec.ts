import compiler from './compiler'

test('compiler svg', async () => {
  const stats = await compiler('./icons/check.svg?react')
  const output = stats.toJson({source: true})?.modules?.[0]?.modules?.[0].source
  expect(output).toMatchSnapshot()
})
