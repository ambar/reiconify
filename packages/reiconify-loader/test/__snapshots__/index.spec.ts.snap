// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler svg 1`] = `
"var __assign = Object.assign;
import React from "react";
import SVG from "base-icon";
function Icon(props) {
  return /* @__PURE__ */ React.createElement(SVG, __assign({
    width: "24",
    height: "24",
    viewBox: "0 0 24 24"
  }, props), /* @__PURE__ */ React.createElement("path", {
    d: "M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  }));
}
var stdin_default = Icon;
export {
  stdin_default as default
};
"
`;

exports[`compiler svg for RN with base-icon 1`] = `
"var __assign = Object.assign;
import React from "react";
import SVGBase from "base-icon";
import * as svg from "react-native-svg";
function Icon(props) {
  return /* @__PURE__ */ React.createElement(SVGBase, __assign({
    width: "24",
    height: "24",
    viewBox: "0 0 24 24"
  }, props), /* @__PURE__ */ React.createElement(svg.Path, {
    d: "M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  }));
}
export {
  Icon as default
};
"
`;

exports[`compiler svg for RN with default 1`] = `
"var __assign = Object.assign;
import React from "react";
import SVGBase from "base-icon/native";
import * as svg from "react-native-svg";
function Icon(props) {
  return /* @__PURE__ */ React.createElement(SVGBase, __assign({
    width: "24",
    height: "24",
    viewBox: "0 0 24 24"
  }, props), /* @__PURE__ */ React.createElement(svg.Path, {
    d: "M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  }));
}
export {
  Icon as default
};
"
`;

exports[`compiler svg for RN with empty base 1`] = `
"var __assign = Object.assign;
import React from "react";
import * as svg from "react-native-svg";
function Icon(props) {
  return /* @__PURE__ */ React.createElement(svg.Svg, __assign({
    width: "24",
    height: "24",
    viewBox: "0 0 24 24"
  }, props), /* @__PURE__ */ React.createElement(svg.Path, {
    d: "M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  }));
}
export {
  Icon as default
};
"
`;
